File: library/network/src/main/java/com/example/network/utils/RequestUtil.java
package com.example.network.utils;

import androidx.annotation.NonNull;

import com.example.network.callback.IResponseCallback;

/**
 * 请求工具类
 */
public class RequestUtil {

    public static String getResponseCallbackName(@NonNull IResponseCallback responseCallback) {
        String name = "unknown";
        try {
            name = responseCallback.toString();
            name = name.substring(name.lastIndexOf(".")+1);
            int aIndex = name.indexOf("@");
            if (aIndex != -1) {
                name = name.substring(0, name.indexOf("@"));
            }
            int sIndex = name.lastIndexOf("$");
            if (sIndex != -1) {
                name = name.substring(0, sIndex);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return name;
    }

    public static String getClassName(@NonNull Object object) {
        String name = "unknown";
        try {
            name = object.getClass().getName();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return name;
    }

    public static String getClassSimpleName(@NonNull Object object) {
        String name = null;
        try {
            name = object.getClass().getSimpleName();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return name;
    }

}

File: library/network/src/main/java/com/example/network/utils/RequestUtil.kt
package com.example.network.utils

import com.example.network.callback.IResponseCallback

/**
 * 请求工具类
 */
object RequestUtil {

    fun getResponseCallbackName(responseCallback: IResponseCallback<*>): String {
        var name = "unknown"
        try {
            name = responseCallback.toString()
            name = name.substring(name.lastIndexOf(".") + 1)
            val aIndex = name.indexOf("@")
            if (aIndex != -1) {
                name = name.substring(0, name.indexOf("@"))
            }
            val sIndex = name.lastIndexOf("$")
            if (sIndex != -1) {
                name = name.substring(0, sIndex)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return name
    }

    fun getClassName(`object`: Any): String {
        var name = "unknown"
        try {
            name = `object`.javaClass.name
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return name
    }

    fun getClassSimpleName(`object`: Any): String? {
        var name: String? = null
        try {
            name = `object`.javaClass.simpleName
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return name
    }
}